cognite/client/data_classes/_base.py

```diff
@@ -281,14 +280,31 @@ def get(self, id: Optional[int] = None, external_id: Optional[str] = None) -> Op
             return self._id_to_item.get(id)
         return self._external_id_to_item.get(external_id)
 
-    def to_pandas(self, camel_case: bool = False) -> pandas.DataFrame:
+    def to_pandas(
+        self,
+        camel_case: bool = False,
+        expand_metadata: bool = False,
+        metadata_prefix: str = "metadata.",
+    ) -> pandas.DataFrame:
         """Convert the instance into a pandas DataFrame.
 
+        Args:
+            camel_case (bool): Convert column names to camel case (e.g. `externalId` instead of `external_id`)
+            expand_metadata (bool): Expand the metadata column into separate columns.
+            metadata_prefix (str): Prefix to use for metadata columns.
+
         Returns:
             pandas.DataFrame: The dataframe.
         """
         pd = cast(Any, utils._auxiliary.local_import("pandas"))
         df = pd.DataFrame(self.dump(camel_case=camel_case))
+
+        if expand_metadata and "metadata" in df.columns:
+            meta_df = pd.json_normalize(df["metadata"]).add_prefix(metadata_prefix)
+            if common_cols := set(meta_df.columns).intersection(set(df.columns)):
+                raise ValueError(f"Metadata contains columns that are already present in the dataframe: {common_cols}")
+            df = pd.concat([df.drop("metadata", axis=1), meta_df], axis=1)
```

Comment:
Is this faster because there is no prefix added to the metadata? I guess that would be an O(n) operation, if we let it fail (since we can do that with `join`), then we can keep the speed benefit. Right? This makes sense to me, but then I need to make a minor adjustment to the function and its test.

Reply:
In the code you have Håkon you lose the prefix. I guess the full code would be

```python
        if expand_metadata and "metadata" in df.columns:
            meta_series = (meta := df.pop("metadata"))[meta.notna()]
            meta_df = pd.DataFrame(meta_series.values.tolist(), index=meta_series.index).add_prefix(metadata_prefix)
            df = df.join(meta_df)
```

Here I also remove the error check, as `join` takes care of it.

Reply:
@doctrino My question above is wondering whether the lost prefix was a deliberate choice/the source of the speedup. Maybe I am bad at wording things, but the prefix insertion is a linear operation, so maybe that is where the speed gains are happening. We could also just assume the end-user will not have issues with overlapping columns. It will make the code faster by not having the prefixes, or maybe doing a check in the code for the length of the prefix, so that it does not unnecessarily do the prefix operation.

Reply:
Yeah I dropped it from both the old and new to simplify the runtime examples as it has ~no effect